File: composeApp/src/androidMain/kotlin/com/calypsan/listenup/client/features/library/LibraryTab.kt
package com.calypsan.listenup.client.features.library

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.outlined.MenuBook
import androidx.compose.material.icons.outlined.AutoStories
import androidx.compose.material.icons.outlined.Person
import androidx.compose.material.icons.outlined.RecordVoiceOver
import androidx.compose.ui.graphics.vector.ImageVector

/**
 * Tabs for the Library screen.
 *
 * Each tab displays a different entity type from the user's collection:
 * - Books: Grid of all audiobooks
 * - Series: Series with book counts
 * - Authors: Authors with book counts
 * - Narrators: Narrators with book counts
 */
enum class LibraryTab(
    val title: String,
    val icon: ImageVector
) {
    Books(
        title = "Books",
        icon = Icons.AutoMirrored.Outlined.MenuBook
    ),
    Series(
        title = "Series",
        icon = Icons.Outlined.AutoStories
    ),
    Authors(
        title = "Authors",
        icon = Icons.Outlined.Person
    ),
    Narrators(
        title = "Narrators",
        icon = Icons.Outlined.RecordVoiceOver
    )
}

File: iosApp/iosApp/Features/Library/LibraryTab.swift
import SwiftUI

/// Tabs available in the Library screen.
///
/// Each tab represents a different view of the user's audiobook collection.
/// Matches Android's `LibraryTab` enum structure.
enum LibraryTab: String, CaseIterable, Identifiable {
    case books
    case series
    case authors
    case narrators

    var id: String { rawValue }

    /// Display title for the tab
    var title: String {
        switch self {
        case .books: String(localized: "library.books")
        case .series: String(localized: "common.series")
        case .authors: String(localized: "library.authors")
        case .narrators: String(localized: "library.narrators")
        }
    }

    /// SF Symbol icon for the tab
    var icon: String {
        switch self {
        case .books: "book.fill"
        case .series: "books.vertical.fill"
        case .authors: "person.fill"
        case .narrators: "waveform.circle.fill"
        }
    }
}
